### ISSUE TITLE
drone-ui does not work in Firefox 49

### ISSUE THREAD

**Issue by orivej** [drone/drone-ui#59]
Loading main page fails with `TypeError: can't prevent extensions on this proxy object`. This happens when `src/actions/tree.js` attemps to create a Baobab tree with a `document.location` leaf, and Baobab attemps to freeze it, which is now prohibited as can be tested with `Object.freeze(document.location)`.

This is likely to require a fix in Baobab, but drone-ui should track it as well.

**Comment by bradrydzewski** [comment]
maybe instead we do something like this to copy the window.location object:
https://github.com/drone/drone-ui/blob/master/src/actions/tree.js#L7

```
// loads the window location from the browser.
let location = window && window.location ? {
  host: window.location.host,
  hostname: window.location.hostname,
  href: window.location.href,
  origin: window.location.origin,
  pathname: window.location.pathname,
  port: window.location.port,
  protocol: window.location.protocol
} : {}
```

**Comment by orivej** [comment]
I believe https://github.com/drone/drone-ui/blob/0273f34c/src/pages/repository/toolbar.js#L25 expects `location.pathname` to update itself between renders, and this seems to be the only place where `tree.location` is used. Maybe remove location from the tree and just use `window.location`?

**Comment by anton-dutov** [comment]
Same issue Firefox 49.0.1/Linux

**Comment by bradrydzewski** [comment]
>  Maybe remove location from the tree and just use window.locatio

the benefit of storing location in the tree is that we can (eventually) use jest for unit tests without having to rely on a browser, using only the virtual dom.

**Comment by bradrydzewski** [comment]
this is fixed by 9b40b6986db7fc986f0d9c1b3aa9adea7e423164. The next drone build will include the updated user interface

**Comment by orivej** [comment]
Well, with your fix Firefox 49 would render drone-ui, but you disabled active tab highlighting: https://github.com/drone/drone-ui/blob/0273f34c/src/pages/repository/toolbar.js#L25 used `location.pathname` to mark the current tab as `active`, but now `location.pathname` is frozen at the moment of tree creation, and no tab is ever marked `active`.
